openedx/features/job_board/tests/views/test_job_board.py

```diff
@@ -53,19 +72,154 @@ def test_create_view_post_unsuccessful(self):
+        self.assertEqual(response_second_page.status_code, status.HTTP_200_OK)
+        self.assertTrue('is_paginated' in response_second_page.context_data)
+        self.assertTrue(response_second_page.context_data['is_paginated'] is True)
+        self.assertTrue(len(response_second_page.context_data['job_list']) == self.number_of_jobs-self.page_length)
+
+    def test_job_list_view_filters_job_type(self):
+        number_of_remote_jobs = 0
+        number_of_onsite_jobs = 0
+
+        for job in self.jobs:
+            if job.type is JOB_TYPE_REMOTE_KEY:
+                number_of_remote_jobs += 1
+            if job.type is JOB_TYPE_ONSITE_KEY:
+                number_of_onsite_jobs += 1
+
+        response = self.client.get(reverse('job_list') + '?' + JOB_TYPE_REMOTE_KEY + '=' + JOB_PARAM_TRUE_VALUE)
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertTrue(response.context_data['search_fields'][JOB_TYPE_REMOTE_KEY], JOB_PARAM_TRUE_VALUE)
+        self.assertTrue(response.context_data['filtered'], True)
+        self.assertTrue(len(response.context_data['job_list']) == number_of_remote_jobs)
+        for job in response.context_data['job_list']:
+            self.assertEqual(job.type, JOB_TYPE_REMOTE_KEY)
+
+        response = self.client.get(reverse('job_list') + '?' + JOB_TYPE_ONSITE_KEY + '=' + JOB_PARAM_TRUE_VALUE)
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertTrue(response.context_data['search_fields'][JOB_TYPE_ONSITE_KEY], JOB_PARAM_TRUE_VALUE)
+        self.assertTrue(response.context_data['filtered'], True)
+        self.assertTrue(len(response.context_data['job_list']) == number_of_onsite_jobs)
+        for job in response.context_data['job_list']:
+            self.assertEqual(job.type, JOB_TYPE_ONSITE_KEY)
+
+    def test_job_list_view_filters_job_compensation(self):
+        number_of_volunteer_jobs = 0
+        number_of_hourly_jobs = 0
+        number_of_salaried_jobs = 0
+
+        for job in self.jobs:
+            if job.compensation is JOB_COMP_VOLUNTEER_KEY:
+                number_of_volunteer_jobs += 1
+            if job.compensation is JOB_COMP_HOURLY_KEY:
+                number_of_hourly_jobs += 1
+            if job.compensation is JOB_COMP_SALARIED_KEY:
+                number_of_salaried_jobs += 1
+
+        response = self.client.get(reverse('job_list') + '?' + JOB_COMP_VOLUNTEER_KEY + '=' + JOB_PARAM_TRUE_VALUE)
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertTrue(response.context_data['search_fields'][JOB_COMP_VOLUNTEER_KEY], JOB_PARAM_TRUE_VALUE)
+        self.assertTrue(response.context_data['filtered'], True)
+        self.assertTrue(len(response.context_data['job_list']) == number_of_volunteer_jobs)
+        for job in response.context_data['job_list']:
+            self.assertEqual(job.compensation, JOB_COMP_VOLUNTEER_KEY)
```

Comment:
Please review this test as well, same issue. Technically test passed but this `assertEqual` never called neither called for other types of `job.compensation` below